src/components/Frame/tests/Frame.test.tsx

```diff
@@ -41,196 +41,241 @@ describe('<Frame />', () => {
-    expect(frame.find('[data-has-navigation]')).toHaveLength(1);
-  });
+    it('renders a CSSTransition around the navigation with an `in` prop set to false if showMobileNavigation is true', () => {
+      const navigation = <div />;
+      const cssTransition = mountWithAppProvider(
+        <Frame showMobileNavigation={false} navigation={navigation} />,
+      )
+        .find(Frame)
+        .find(TrapFocus)
+        .find(CSSTransition);
 
-  it('does not render with a has nav data attribute when nav is not passed', () => {
-    const frame = mountWithAppProvider(<Frame />);
-    expect(frame.find('[data-has-navigation]')).toHaveLength(0);
-  });
+      expect(cssTransition.prop('in')).toBe(false);
+    });
 
-  it('renders with a top bar data attribute if a topBar is passed', () => {
-    const topbar = <div />;
-    const topBar = mountWithAppProvider(<Frame topBar={topbar} />);
+    it('renders a CSSTransition around the navigation with an `in` prop set to true if showMobileNavigation is true', () => {
+      const navigation = <div />;
+      const cssTransition = mountWithAppProvider(
+        <Frame showMobileNavigation navigation={navigation} />,
+      )
+        .find(Frame)
+        .find(TrapFocus)
+        .find(CSSTransition);
 
-    expect(topBar.find('[data-polaris-top-bar]')).toHaveLength(1);
-  });
+      expect(cssTransition.prop('in')).toBe(true);
+    });
+
+    it('renders with a has nav data attribute when nav is passed', () => {
+      const navigation = <div />;
+      const frame = mountWithAppProvider(<Frame navigation={navigation} />);
+      expect(frame.find('[data-has-navigation]')).toHaveLength(1);
+    });
 
-  it('does not render with a top bar data attribute if none is passed', () => {
-    const topBar = mountWithAppProvider(<Frame />);
+    it('does not render with a has nav data attribute when nav is not passed', () => {
+      const frame = mountWithAppProvider(<Frame />);
+      expect(frame.find('[data-has-navigation]')).toHaveLength(0);
+    });
 
-    expect(topBar.find('[data-polaris-top-bar]')).toHaveLength(0);
-  });
+    it('does not call onNavigationDismiss when escape is pressed and screen size is large', () => {
+      const spy = jest.fn();
+      const frame = mountWithApp(
+        <Frame navigation={<div />} onNavigationDismiss={spy} />,
+      );
+      frame
+        .find('div', {id: 'AppFrameNav'})!
+        .trigger('onKeyDown', {key: 'Escape'});
 
-  // JSDOM 11.12.0 does not support setting/reading custom properties so we are
-  // unable to assert that we set a custom property
-  // See https://github.com/jsdom/jsdom/issues/1895
-  // eslint-disable-next-line jest/no-disabled-tests
-  it.skip('sets a root property with global ribbon height if passed', () => {
-    mountWithAppProvider(<Frame globalRibbon={<div />} />);
-    expect(documentHasStyle('--global-ribbon-height', '0px')).toBe(true);
-  });
+      const {onNavigationDismiss} = frame.props as FrameProps;
```

Comment:
Can we avoid this casting?

Reply:
Not that I know of. I think casting is ok in tests? 

The reason I think we need casting here is because this component is an older class component and I'm using the new test utils on it and thus can't infer the prop types. Open to trying other suggestions though

Reply:
It seems the type signature for root-level props are inferred at `root.prop(x: never)`. I'm guessing it's probably intentional to discourage the use of prop manipulation and such on the root, e.g

```tsx
const spy = jest.fn();
const frame = mountWithApp(<Frame navigation={<div /> onNavigationDismiss={spy}} />
frame.trigger('onNavigationDismiss'); // frame.prop('onNavigationDismiss')();
expect(spy).toHaveBeenCalled(1);

// is essentially
const spy = jest.fn();
mountWithApp(<Frame navigation={<div /> />
spy();
expect(spy).toHaveBeenCalled(1);
```

However, in this situation we can omit this line.
```suggestion
```